kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -477,3 +498,50 @@ fn collect_alloc_items<'tcx>(tcx: TyCtxt<'tcx>, alloc_id: AllocId) -> Vec<MonoIt
     };
     items
 }
+
+/// This MIR Visitor is intended for one specific purpose:
+/// Find the closure that exist inside a top-level const declaration generated by
+/// test declarations. This allows us to treat this closure instance as a root for
+/// the reachability analysis.
+///
+/// Entry into this visitor will be via `visit_body`
+struct ConstMonoItemExtractor<'a, 'tcx> {
+    tcx: TyCtxt<'tcx>,
+    collected: FxHashSet<MonoItem<'tcx>>,
+    body: &'a Body<'tcx>,
+}
+
+impl<'a, 'tcx> MirVisitor<'tcx> for ConstMonoItemExtractor<'a, 'tcx> {
+    fn visit_rvalue(&mut self, rvalue: &Rvalue<'tcx>, location: Location) {
+        trace!(rvalue=?*rvalue, "visit_rvalue");
+
+        match *rvalue {
+            Rvalue::Cast(CastKind::Pointer(PointerCast::ClosureFnPointer(_)), ref operand, _) => {
+                let source_ty = operand.ty(self.body, self.tcx);
+                // Hazard: We don't have an instance, so there's no monomorphize possible here.
```

Comment:
I'm not sure what you mean. Do you mean applying `Instance::mono` to the `const`'s `DefId`?

Possibly yes, I didn't think of that, so I haven't tried it. Would you prefer the code be written that way? (With this visitor having an `Instance` added to the struct so this could be written I guess "more conventionally")

Reply:
I went ahead and switched over to doing what I think you were suggesting.